spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -814,8 +822,19 @@ struct ProductionVisitor
             pb->saveParsePosition();
         }
 
-        if ( ncur )
+        else if ( ncur_max_size ) {
+            // Compute how far to advance for `&max-size` parsing where we operate on a limited view, but do not
+            // necessarily consume it fully. Since `cur` and `ncur_max_size` point to different views we need compute
+            // the difference in offset; this is safe since the limited view is into the original stream `cur` points
+            // to.
+            ncur = builder::memberCall(state().cur, "advance",
+                                       {builder::difference(builder::memberCall(*ncur_max_size, "offset", {}),
+                                                            builder::memberCall(state().cur, "offset", {}))});
+        }
```

Comment:
```suggestion
        else if ( ncur_max_size )
            // Compute how far to advance for `&max-size` parsing where we operate on a limited view, but do not
            // necessarily consume it fully. Since `cur` and `ncur_max_size` point to different views we need compute
            // the difference in offset; this is safe since the limited view is into the original stream `cur` points
            // to.
            ncur = builder::memberCall(state().cur, "advance",
                                       {builder::difference(builder::memberCall(*ncur_max_size, "offset", {}),
                                                            builder::memberCall(state().cur, "offset", {}))});
```